Nil pointer panic in failsTokenEfficiencyFilter when tiktoken download fails

## Description

When using betterleaks as a Go library, `DetectString()` panics with a nil pointer dereference if the tiktoken `cl100k_base` tokenizer fails to download or parse. The `failsTokenEfficiencyFilter` method calls `tiktoken.Encode()` on a nil `*Tiktoken` receiver without checking initialization status.

## Stack trace

```
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV]

github.com/pkoukk/tiktoken-go.(*Tiktoken).Encode(0x0, ...)
    tiktoken.go:81
github.com/betterleaks/betterleaks/detect.(*Detector).failsTokenEfficiencyFilter(...)
    detect.go:755
github.com/betterleaks/betterleaks/detect.(*Detector).detectRule(...)
    detect.go:727
github.com/betterleaks/betterleaks/detect.(*Detector).DetectContext(...)
    detect.go:482
github.com/betterleaks/betterleaks/detect.(*Detector).DetectString(...)
    detect.go:291
```

## Root cause

`tiktoken-go` fails to load the `cl100k_base` tokenizer with a parse error:

```
WRN Could not pull down cl100k_base tiktokenizer
    error="strconv.Atoi: parsing \"43416\\x1aLmRldGFpbHM=\": invalid syntax"
```

When this happens, the `*Tiktoken` value is nil. `failsTokenEfficiencyFilter` at `detect.go:755` calls `Encode()` on it without a nil check, causing the panic.

## Reproduction

```go
package main

import "github.com/betterleaks/betterleaks/detect"

func main() {
    d, _ := detect.NewDetectorDefaultConfig()
    // Panics if tiktoken download fails (e.g., network issues, corrupted cache)
    _ = d.DetectString("secret: AKIAQWERTYUIOP3B5TFD")
}
```

The panic is more reliably triggered with `go test -race` since parallel goroutines expose a race condition in the lazy tiktoken initialization.

## Suggested fix

Add a nil check before calling `Encode()` in `failsTokenEfficiencyFilter`:

```go
func (d *Detector) failsTokenEfficiencyFilter(match string) bool {
    if d.tiktokenizer == nil {
        return false // skip filter if tokenizer unavailable
    }
    // ... existing logic
}
```

This would gracefully degrade to skipping token-efficiency scoring when the tokenizer is unavailable, rather than panicking.

## Environment

- betterleaks v1.1.1
- Go 1.26.1 linux/amd64
- Using betterleaks as a library (not CLI)

--- a/detect/detect.go
+++ b/detect/detect.go
@@ -748,6 +748,12 @@ func (d *Detector) detectRule(fragment sources.Fragment, currentRaw string, r co
 }
 
 func (d *Detector) failsTokenEfficiencyFilter(secret string) bool {
+	// Skip token-efficiency filtering if the tokenizer failed to initialize.
+	// (e.g., network error downloading cl100k_base)
+	if d.tokenizer == nil {
+		return false
+	}
+
 	// For short secrets (< 20 chars) that contain newlines, strip the newlines
 	// before analysis so that strings like "123\n\nTest" are evaluated as "123Test"
 	// allowing word detection to work.

